cirq-core/cirq/transformers/routing/mapping_manager.py

```diff
@@ -133,3 +136,12 @@ def shortest_path(self, lq1: 'cirq.Qid', lq2: 'cirq.Qid') -> Sequence['cirq.Qid'
     @cached_method
     def _physical_shortest_path(self, pq1: 'cirq.Qid', pq2: 'cirq.Qid') -> Sequence['cirq.Qid']:
         return nx.shortest_path(self._induced_subgraph, pq1, pq2)
+
+    def _value_equality_values_(self):
+        return self.graph_adjacency, self._map
+
+    def __str__(self) -> str:
+        return f'Device graph adjacency: {self.graph_adjacency}\nMap: {self._map}'
+
+    def __repr__(self) -> str:
+        return f'cirq.MappingManager(nx.Graph({self.graph_adjacency}), {self._map})'
```

Comment:
It would be nice if we could just use the `repr()`s of the constituent objects but other packages don't use reprs as a serialization format as long time attendees of the cirq cync will remember me complaining about our convention. 

Anyways, I think the proper cirq thing to do is to use `cirq._compat.proper_repr` but I don't remember if that supports networkx graphs